articles/151_roslaunch_xml.md

```diff
@@ -0,0 +1,223 @@
+{{ page.abstract }}
+</div>
+
+Authors: {{ page.author }}
+
+
+# ROS 2 Launch XML Format v0.1.0
+
+## Rationale
+
+As an alternative to a programmatic approach to the ROS 2 launch system's API, a declarative description features a WYSIWYG approach, easier to read, audit and maintain.
+This is the preferred approach for ROS 1 `roslaunch` launch files, thus some degree of familiarity is expected (and relied upon).
+
+The body of such a description is mainly comprised of statically declared launch actions with a prescribed configuration.
+To that, runtime value substitution is added in order to fullfill common dynamic (re)configuration needs like conditional execution, resource lookups, etc.
+It is intended for these entities to map to those of the underlying implementation, reducing support to file parsing.
+
+This article describes XML aiming to ease the bridge between ROS and ROS 2 launch files.
+YAML is currently supported too, and other markup languages could be added.
+
+## Static Description
+
+### Schema Definition
+
+```xml
+{% include_relative specs/launch.0.1.0.xsd %}
+```
+
+### Tags Semantics
+
+#### All Tags
+
+Every tag's execution can be conditioned on a boolean predicate via `if` or `unless` attributes.
+All action tags that can contain other action tags can scope ROS entities in a namespace via the `ns` attribute.
+
+#### `<launch>` Tag
+
+Root tag of any launch file.
+There must only be one `<launch>` tag on a given launch file.
+
+#### `<include>` Tag
+
+##### Description
+
+The `<include>` tag allows for bringing a launch file description into another, enabling re-use of hierarchical launch layouts.
+The included launch file description has its own scope for launch configurations.
+The included launch file description is not necessarily written in this format nor a declarative one (i.e. a programmatic description).
+To avoid ROS name clashes, included nodes can be namespaced via the `ns` attribute.
+
+##### Examples
+
+```xml
+<include file="/opt/my_launch_file.py" ns="/my_ns"/>
+<include file="$(find-pkg my_pkg)/launch/some_launch_file.xml"/>
+<include file="/opt/my_other_launch_file.xml">
+  <arg name="some_argument" value="dummy_value"/>
+</include>
+```
+
+#### `<group>` Tag
+
+##### Description
+
+The `<group>` tag allows for launch actions' grouping as well as optional launch file configuration scoping.
+
+##### Examples
+
+```xml
+<group ns="dummy_group" scoped="true">
+  <node package="a_ros_package" name="dummy0" executable="dummy_node"/>
+  <node package="a_ros_package" name="dummy1" executable="dummy_node"/>
+</group>
+```
+
+Namespaces for groups are usually relative to allow further namespacing by other launch files including the one the group is in.
+
+#### `<let>` Tag
+
+##### Description
+
+The `<let>` tags allows for definition of scoped launch file configuration variables.
+
+##### Examples
+
+```
+<let var="foo" value="$(env BAR)"/>
+<let var="baz" value="false"/>
+```
+
+#### `<arg>` Tag
```

Comment:
I forgot implementing this one, will do soon.

Reply:
I think this is an important one to do 👍.

Reply:
Done in https://github.com/ros2/launch/pull/270